[RHDHBUGS-2722]: Restructure authentication title to JTBD outline (#2049)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-enable-authentication-with-rhbk.adoc

:_mod-docs-content-type: PROCEDURE

[id="enable-authentication-with-rhbk_{context}"]
= Enable authentication with {rhbk-brand-name} ({rhbk})

[role="_abstract"]
Configure {rhbk-brand-name} ({rhbk}) as your {product} sign-in provider by enabling the OIDC authentication provider.

.Prerequisites
* You have xref:share-a-secret-with-rhbk_{secrets-context}[shared a secret with {rhbk}].

.Procedure

> . The OIDC provider authentication backend plugin requires {product-short} to support sessions.

Reviewer: technically, the explanation why you need to enable session support isn't necessary in a procedure step. I'd turn it into a note

Enable session support by adding the session secret to your `{my-app-config-file}` file:
+
[source,yaml]
----
auth:
  session:
    secret: ${SESSION_SECRET}
----

> . Enable the {rhbk} authentication provider, by adding the OIDC provider section in your  `{my-app-config-file}` file:

Reviewer: ```suggestion
. Add an OIDC provider section to your  `{my-app-config-file}` file:
```

. From the *Realm Settings* page, click the *Tokens* tab.
. From the *Refresh tokens* section of the *Tokens* tab, toggle the *Revoke Refresh Token* to the *Enabled* position.
====

> . To disable the guest login option, in the `{my-app-config-file}` file, set the authentication environment to `production`:

Reviewer: is this optional?

Author: See above

+
[source,yaml]
----
auth:
  environment: production
----

.Verification
. To verify {rhbk} user authentication:
.. Go to the {product-short} login page.
.. Your {product-short} sign-in page displays *Sign in using OIDC* and the Guest user sign-in is disabled.

> .. Log in with OIDC by using the saved **Username** and **Password** values.

Reviewer: ```suggestion
. Go to the {product-short} login page.
. Your {product-short} sign-in page displays *Sign in using OIDC* and the Guest user sign-in is disabled.
. Log in with OIDC by using the saved **Username** and **Password** values.
```
